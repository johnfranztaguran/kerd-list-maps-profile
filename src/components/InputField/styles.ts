import { StyleSheet, ViewStyle, TextStyle } from 'react-native'
import { dimens, fonts, colors, platform } from 'constant'

interface Styles {
  wrapper: ViewStyle
  inputContainer: ViewStyle
  input: ViewStyle
  prefix: TextStyle
  helper: ViewStyle
  info: TextStyle
}
const { isAndroid } = platform

const styles: Styles = StyleSheet.create<Styles>({
  wrapper: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  inputContainer: {
    flexGrow: 1
  },
  input: {
    height: 40
  },
  prefix: {
    lineHeight: 56,
    height: 56,
    fontFamily: fonts.OpenSansRegular,
    fontWeight: '500',
    fontSize: dimens.fontSize.size_16,
    paddingTop: 4,
    paddingRight: 14
  },
  helper: {
    paddingLeft: 56
  },
  info: {
    color: colors.gunPowder,
    fontFamily: isAndroid ? fonts.OpenSansSemibold : fonts.OpenSansRegular,
    fontWeight: '600'
  }
})

export default styles
