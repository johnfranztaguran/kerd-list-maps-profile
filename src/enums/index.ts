export enum Screens {
  Main = 'Main',
  Login = 'Login',
  Register = 'Register',
  CompleteRegister = 'CompleteRegister',
  Welcome = 'Welcome',
  ListForm = 'ListForm',
  ProfileSettings = 'ProfileSettings',
  GoogleMaps = 'GoogleMaps',
  ListCrud = 'ListCrud',
  ProfileCrud = 'ProfileCrud'
}

export enum TextInputType {
  Number = 'number',
  Decimal = 'decimal',
  Text = 'text',
  Email = 'email',
  Password = 'password'
}

export enum MessageType {
  Error = 'error',
  Info = 'info'
}

export enum ErrorType {
  Empty = 'empty',
  Invalid = 'invalid',
  Existed = 'existed',
  Info = 'info',
  InvalidCombination = 'invalidCombination'
}

export enum HelperMessageType {
  Info = 'info',
  Error = 'error'
}

export enum ButtonMode {
  Outlined = 'outlined',
  Contained = 'contained',
  Text = 'text'
}

export enum ButtonType {
  PlainButton = 'button',
  ButtonIcon = 'button-icon'
}
