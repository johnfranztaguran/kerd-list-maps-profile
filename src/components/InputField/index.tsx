import React from 'react'
import { View, KeyboardTypeOptions, Text } from 'react-native'
import { TextInput, HelperText } from 'react-native-paper'
import { RenderProps } from 'react-native-paper/lib/typescript/src/components/TextInput/types'
import { testIdProps } from 'utils/testAutomation'
import { HelperMessageType } from 'enums'
import { ErrorMessage } from 'types'
import { strings } from 'constant'
import styles from './styles'

type Props = {
  helperMessage?: ErrorMessage
  error?: boolean
  mode?: 'flat' | 'outlined'
  label?: string
  value?: string
  autoFocus?: boolean
  keyboardType?: KeyboardTypeOptions
  placeholder?: string
  contextMenuHidden?: boolean
  autoCorrect?: boolean
  blurOnSubmit?: boolean
  editable?: boolean
  allowFontScaling?: boolean
  disabled?: boolean
  pointerEvents?: any
  style?: any
  theme?: any
  prefix?: string
  onChangeText?: (value: string) => void
  render?: (props: RenderProps) => React.ReactNode
  onSubmitEditing?: () => void
  inputReference?: (ref: any) => void
}

const InputField = (props: Props) => {
  const {
    error = false,
    helperMessage = { type: HelperMessageType.Error, message: '' },
    mode = 'outlined',
    label,
    value,
    autoFocus,
    keyboardType,
    placeholder,
    onChangeText,
    render,
    onSubmitEditing,
    contextMenuHidden,
    autoCorrect,
    blurOnSubmit,
    editable,
    disabled,
    pointerEvents,
    theme,
    style,
    prefix,
    inputReference
  } = props

  const isErrorType = helperMessage.type === HelperMessageType.Error

  return (
    <View>
      <View style={styles.wrapper}>
        {prefix && (
          <Text
            {...testIdProps(strings.testIdText.prefixTxt + prefix)}
            style={styles.prefix}>
            {prefix}
          </Text>
        )}
        <View style={styles.inputContainer}>
          <TextInput
            accessibilityStates={undefined}
            accessibilityComponentType={undefined}
            accessibilityTraits={undefined}
            autoCompleteType={undefined}
            {...testIdProps(strings.testIdText.prefixInput + label)}
            style={[style, styles.input]}
            error={isErrorType && error}
            mode={mode}
            label={label}
            value={value}
            keyboardType={keyboardType}
            placeholder={placeholder}
            onChangeText={onChangeText}
            autoFocus={autoFocus}
            autoCorrect={autoCorrect}
            blurOnSubmit={blurOnSubmit}
            autoCapitalize="none"
            render={render}
            editable={editable}
            disabled={disabled}
            allowFontScaling
            theme={theme}
            pointerEvents={pointerEvents}
            contextMenuHidden={contextMenuHidden}
            onSubmitEditing={onSubmitEditing}
            ref={inputReference}
          />
        </View>
      </View>

      {error && (
        <HelperText
          {...testIdProps(`${strings.testIdText.prefixTxt} HelperText`)}
          type={helperMessage.type}
          visible={error}
          style={[
            prefix && styles.helper,
            helperMessage.type === HelperMessageType.Info ? styles.info : null
          ]}>
          {helperMessage.message}
        </HelperText>
      )}
    </View>
  )
}

export default InputField
