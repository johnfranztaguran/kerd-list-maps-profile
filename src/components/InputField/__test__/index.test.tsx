import React from 'react'
import { render } from '@testing-library/react-native'
import { HelperMessageType } from 'enums'
import InputField from '../index'

describe('Component: InputField', () => {
  it('Should render a InputField', () => {
    const { baseElement } = render(<InputField />)
    expect(baseElement).toMatchSnapshot()
  })

  it('Should render a InputField with error message', () => {
    const errorMsg = 'Invalid input'
    const helperMessage = { type: HelperMessageType.Error, message: errorMsg }
    const { baseElement, getByText } = render(
      <InputField error helperMessage={helperMessage} />
    )

    const element = getByText(errorMsg)
    expect(element.children[0]).toEqual(errorMsg)
    expect(baseElement).toMatchSnapshot()
  })
})
