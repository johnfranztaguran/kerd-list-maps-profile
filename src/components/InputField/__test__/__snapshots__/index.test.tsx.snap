// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: InputField Should render a InputField 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                Object {},
              ]
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 40,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <TextInput
                  accessibilityLabel="input-undefined"
                  accessible={true}
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  multiline={false}
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  selectionColor="#6200ee"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "textAlign": "left",
                        "zIndex": 1,
                      },
                      Object {
                        "height": 40,
                      },
                      Object {
                        "paddingBottom": 12,
                        "paddingTop": 12,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlignVertical": "center",
                      },
                    ]
                  }
                  testID="input-undefined"
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component: InputField Should render a InputField with error message 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                Object {},
              ]
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "borderColor": "#B00020",
                      "borderRadius": 4,
                      "borderWidth": 2,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 40,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <TextInput
                  accessibilityLabel="input-undefined"
                  accessible={true}
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  multiline={false}
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  selectionColor="#B00020"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "textAlign": "left",
                        "zIndex": 1,
                      },
                      Object {
                        "height": 40,
                      },
                      Object {
                        "paddingBottom": 12,
                        "paddingTop": 12,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlignVertical": "center",
                      },
                    ]
                  }
                  testID="input-undefined"
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        accessibilityLabel="txt--helpertext"
        accessible={true}
        style={
          Object {
            "color": "#B00020",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "400",
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 4,
            "textAlign": "left",
            "transform": Array [
              Object {
                "translateY": -0,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="txt--helpertext"
      >
        Invalid input
      </Text>
    </View>
  </View>
</View>
`;
